posthog.go

```diff
@@ -663,6 +687,19 @@ func (c *client) getFeatureFlagFromDecide(key string, distinctId string, groups
 	return false, nil
 }
 
+func (c *client) getFeatureFlagPayloadFromDecide(key string, distinctId string, groups Groups, personProperties Properties, groupProperties map[string]Properties) (string, error) {
+	decideResponse, err := c.makeDecideRequest(distinctId, groups, personProperties, groupProperties)
+	if err != nil {
+		return "", err
+	}
+
+	if value, ok := decideResponse.FeatureFlagPayloads[key]; ok {
+		return value, nil
+	}
+
+	return "", nil
+}
+
```

Comment:
I see what you mean when you're asking about when to return `"", nil`, since this is an example where you do that.  Currently, the model we've kept for this SDK (that I've sure you've seen) is to have all of these methods return `interface{}, error`, instead of something with better types.  The original decision was before my time, but I think the idea was probably to flexibly handle the fact that a feature flag could either be true, false, or a string.

However, now that we're venturing into handling payloads, I'm fine with this approach of more explicit typing, since the `featureFlagPayload` values should always be strings.

Since you're the one implementing this new feature (no one else is using payloads in the Go SDK yet), I think it's up to you – if you want to work with feature flag payload values as strings everywhere, then we can keep it like this.  Perhaps it's worth calling out the difference between feature flag payload values vs feature flag values (the types do a good job of explaining _what_, but maybe a comment is worth the _why_).  But overall I think this approach is reasonable.

Reply:
Got it! I'm inclined to change the return type from interface{} to string across the board, including in the user-facing method GetFeatureFlagPayload. If I have the opportunity to use explicit types, I'll take it.

The changes I'll be making are:

- Changing the return type from interface{} to string.
- Returning an empty string and a nil error if there was no actual error, but the feature flag doesn't have a payload defined

Reply:
yup, that sounds good to me!